src/run.c

```diff
@@ -716,6 +719,120 @@ Port io_dl_close(Net* net, Book* book, Port argm) {
   return inject_ok(net, new_port(ERA, 0));
 }
 
+// Deletes a single file or an empty directory at the specified path.
+// Returns Ok(None) if successful, or Err(reason) if an error occurs.
+// This function attempts to remove both files and empty directories without
+// first checking the type of the path.
+// Returns: Result<*, IOError<i24>>
+Port io_delete_file(Net* net, Book* book, Port argm) {
+  Str path = readback_str(net, book, argm);
+
+  int result = remove(path.buf);
+  free(path.buf);
+
+  if (result == 0) {
+    return inject_ok(net, new_port(ERA, 0));
+  } else {
+    return inject_io_err_inner(net, new_port(NUM, new_i24(errno)));
+  }
+}
+
+int delete_directory_recursive(const char* path) {
+  DIR *d = opendir(path);
+  size_t path_len = strlen(path);
+  int r = -1;
+
+  if (d) {
+    struct dirent *p;
+    r = 0;
+
+    while (!r && (p = readdir(d))) {
+      int r2 = -1;
+      char *buf;
+      size_t len;
+
+      if (!strcmp(p->d_name, ".") || !strcmp(p->d_name, "..")) {
+        continue;
+      }
+
+      len = path_len + strlen(p->d_name) + 2;
+      buf = malloc(len);
+
+      if (buf) {
+        struct stat statbuf;
+        snprintf(buf, len, "%s/%s", path, p->d_name);
+
+        if (!stat(buf, &statbuf)) {
+          if (S_ISDIR(statbuf.st_mode)) {
+            r2 = delete_directory_recursive(buf);
```

Comment:
This will stack overflow. Prefer using `nftw` or a stack/list of files to remove